packages/material-ui-styles/src/styled/styled.spec.tsx

```diff
@@ -85,10 +84,14 @@ function acceptanceTest() {
       fontFamily: theme.fontFamily,
     }),
   );
+  const StyledInferedPropsMyComponent = styled(MyComponent)(({ defaulted }) => ({
+    content: defaulted,
+  }));
   const renderedMyComponent = (
     <React.Fragment>
       <MyComponent className="test" />
       <StyledMyComponent theme={MyThemeInstance} />
+      <StyledInferedPropsMyComponent defaulted="Hi!" className="test" />
```

Comment:
This is an issue. `className` isn't required. It should pass without it.

Reply:
Have you looked at [source](https://github.com/mui-org/material-ui/blob/master/packages/material-ui-styles/src/styled/styled.d.ts#L31)?

Reply:
I have got about what you.
`className` is required by `MyComponentProps`. Non required `className` is overridden by [this](https://github.com/mui-org/material-ui/blob/master/packages/material-ui-styles/src/styled/styled.d.ts#L28).

Reply:
I have changed it but I am not sure, should we change original props?